src/components/views/elements/Spoiler.js

```diff
@@ -0,0 +1,32 @@
+'use strict';
+
+import React from 'react';
+
+module.exports = React.createClass({
+    displayName: 'Spoiler',
+
+    getInitialState() {
+        return {
+            visible: false,
+        };
+    },
+
+    toggleVisible() {
+        this.setState({ visible: !this.state.visible });
+    },
+
+    render: function() {
+        const reason = this.props.reason ? (
+            <span className="mx_EventTile_spoiler_reason">{"(" + this.props.reason + ")"}</span>
+        ) : null;
+        return (
+            <span className={"mx_EventTile_spoiler" + (this.state.visible ? " visible" : "")} onClick={this.toggleVisible.bind(this)}>
+                { reason }
+                &nbsp;
+                <span className="mx_EventTile_spoiler_content">
+                    <span dangerouslySetInnerHTML={{ __html: this.props.contentHtml }} />
```

Comment:
This is pretty stupid. Soru had to pass the content as HTML as, apparently, react doesn't allow attaching a DOM node as child...does anyone know a cleaner solution?